src/Type/Traits/NonOffsetAccessibleTypeTrait.php

```diff
@@ -14,6 +14,11 @@ public function isOffsetAccessible(): TrinaryLogic
 		return TrinaryLogic::createNo();
 	}
 
+	public function isOffsetAccessLegal(): TrinaryLogic
+	{
+		return TrinaryLogic::createYes();
+	}
```

Comment:
The most important difference between `isOffsetAccessible` is this line. Types like integer, boolean are not offset accessible but doesn't throw a fatal error on access in `isset`.

Reply:
I don't like this being added to a trait with a name like this. I'd like a separate trait for this method that would be used in all the places where relevant. Or maybe no trait at all, and just put this implementation in all the places where it applies.